### PR TITLE
Refactor karton.core.test to mock backend

### LINKED ISSUE #44: KartonTest: mock backend instead of changing base class.
`karton.test` is unittest suite for testing Karton services.

Currently it mocks the Karton classes by forcefully changing their base class from Consumer to KartonMock. Unfortunately this approach doesn't work well, when Karton service implements custom `__init__` method.

With KartonBackend we can just setup a mock on the redis/minio level instead of hacking the Consumer class.

### LINKED ISSUE #79: Refactor KartonTestCase to mock KartonBackend

### PR DESCRIPTION
**Pull request by nazywam**
Closes #44

### CODE REVIEW

File: karton/core/test.py [hunk 1]
```diff
@@ -1,85 +1,91 @@
...
+    ) -> bool:
+        log.debug("Producing a log from [%s]: %s", logger_name, log_record)
+        # Return a truthy value to signal that the message has been consumed
+        return True
+
+    def increment_metrics(self, metric: KartonMetrics, identity: str) -> None:
+        log.debug("Incrementing metric %s for identity %s", metric, identity)
+
+    def remove_object(self, bucket: str, object_uid: str) -> None:
+        del self.buckets[bucket][object_uid]
```
Comment (msm-code) [inline karton/core/test.py:65]: no log debug here and below? (since other methods have ones)

File: karton/core/test.py [hunk 2]
```diff
@@ -215,87 +217,16 @@ def run_task(self, task: Task) -> List[Task]:
...
-        if self.current_task is not None:
-            task.set_task_parent(self.current_task)
-            task.merge_persistent_payload(self.current_task)
-            task.priority = self.current_task.priority
+        self.karton.backend.produced_tasks = []
+        self.karton.current_task = task
 
-        task.last_update = time.time()
-        task.headers.update({"origin": self.identity})
+        if get_function_arg_num(self.karton.process) == 0:
```
Comment (msm-code) [inline karton/core/test.py:223]: is this backward compatiblity for other karton versions?

(Maybe add a small comment, like # backward compatibility for karton <3.0)

Comment (nazywam) [reply karton/core/test.py:223]: Both parameter passing methods are accepted right now, added a comment anyway

**Review by psrok1 — APPROVED** [review]
Looks great :+1:

**Pull request merged**